tests/platform_helper/domain/test_copilot_environment.py

```diff
@@ -259,3 +265,334 @@ def test_when_multiple_https_certificate_present(self, mock_find_https_listener)
 
         certificate_arn = find_https_certificate(boto_mock, "test-application", "development")
         assert "certificate_arn_default" == certificate_arn
+
+
+class TestCrossEnvironmentS3Templating:
+    def environments(self):
+        return {
+            "dev": {"accounts": {"deploy": {"name": "dev-acc", "id": "123456789010"}}},
+            "staging": {"accounts": {"deploy": {"name": "dev-acc", "id": "123456789010"}}},
+            "hotfix": {"accounts": {"deploy": {"name": "prod-acc", "id": "987654321010"}}},
+            "prod": {"accounts": {"deploy": {"name": "prod-acc", "id": "987654321010"}}},
+        }
+
+    def s3_xenv_extensions(self):
+        return {
+            "test-s3-bucket-x-account": {
+                "type": "s3",
+                "services": "test-svc",
+                "environments": {
+                    "hotfix": {
+                        "bucket_name": "x-acc-bucket",
+                        "cross_environment_service_access": {
```

Comment:
Slightly different structure than what's in the [DBTP-1568-s3-cross-env-service-access PR ](https://github.com/uktrade/demodjango-deploy/pull/256/files).

Reply:
That link it to an old PR. This is the current work on that branch: https://github.com/uktrade/demodjango-deploy/blob/DBTP-1568-s3-cross-env-service-access/platform-config.yml (we now allow access to a bucket from multiple other envs).